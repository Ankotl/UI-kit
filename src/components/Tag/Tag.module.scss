@import '../../styles/_typography';
@import '../../styles/tokens/_colors';
@import '../../styles/tokens/_transitions';

.Tag {
  align-items: center;
  border: none;
  border-radius: 100px;
  width: fit-content;
  color: $color-black;
  background: $color-bg;
  transition: color $transition-main, background-color $transition-main;

  &:hover {
    color: $color-original-blue;
  }
}

.Tag_selected {
  color: #fff;
  background-color: $color-original-blue;

  &:hover {
    color: #fff;
  }
}

.Tag_lg {
  @include typography-p1;

  padding: 4px 14px;
}

.Tag_sm {
  @include typography-p3;

  padding: 4px 12px;
}