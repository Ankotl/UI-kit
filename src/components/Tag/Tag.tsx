import React from 'react'

import clsx from 'clsx'

import s from './Tag.module.scss'

export const TagDefaultAsType = 'div' as const
export type TagDefaultAsType = typeof TagDefaultAsType

export interface ITagBaseProps<E extends React.ElementType>
  extends React.HtmlHTMLAttributes<HTMLDivElement> {
  size?: 'sm' | 'lg'
  text?: string
  isSelected?: boolean
  as?: E
}

export type ITagProps<E extends React.ElementType> = ITagBaseProps<E> &
  Omit<React.ComponentProps<E>, keyof ITagBaseProps<E>>

export const Tag = <E extends React.ElementType = 'div'>({
  text,
  className,
  isSelected,
  children,
  as,
  size = 'sm',
  ...otherProps
}: ITagProps<E>) => {
  const Component = as ?? TagDefaultAsType

  return (
    <Component
      {...otherProps}
      className={clsx(
        s.Tag,
        {
          [s.Tag_sm]: size === 'sm',
          [s.Tag_lg]: size === 'lg',
          [s.Tag_selected]: isSelected,
        },
        className
      )}
    >
      {children ?? text}
    </Component>
  )
}
