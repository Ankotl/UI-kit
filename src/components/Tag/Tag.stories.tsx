import React from 'react'

import { getStoryMeta } from '../../utils/storybook/getStoryMeta'
import { getStoryTemplate } from '../../utils/storybook/getStoryTemplate'

import { ITagProps, Tag as TagCmp } from './Tag'

const storiesMeta = getStoryMeta({
  component: TagCmp,
})
export default storiesMeta

const storyTemplate = getStoryTemplate((props: ITagProps<'div'>) => {
  const containerStyle = {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    height: '100vh',
  }

  return (
    <div style={containerStyle}>
      <TagCmp {...props} text={props.text || 'Test text'} />
    </div>
  )
})

export const Tag = storyTemplate({})
