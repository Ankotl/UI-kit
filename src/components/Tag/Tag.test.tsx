import React from 'react'

import { render, screen } from '@testing-library/react'

import { Tag } from './Tag'

describe('Tag component test', () => {
  test('Has right states', () => {
    const { container } = render(<Tag isSelected />)
    expect(container).toMatchSnapshot()

    const { container: containerSecondary } = render(<Tag size="sm" />)
    expect(containerSecondary).toMatchSnapshot()
  })

  test('Renders content', () => {
    const testText = 'testText'
    const testTextSecond = 'testTextSecond'

    render(<Tag text={testText} />)
    render(<Tag>{testTextSecond}</Tag>)

    expect(screen.getByText(testText)).toBeTruthy()
    expect(screen.getByText(testTextSecond)).toBeTruthy()
  })
})
