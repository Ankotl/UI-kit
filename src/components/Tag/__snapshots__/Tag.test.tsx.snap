// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component test Has right states 1`] = `
<div>
  <div
    class="Tag Tag_sm Tag_selected"
  />
</div>
`;

exports[`Tag component test Has right states 2`] = `
<div>
  <div
    class="Tag Tag_sm"
  />
</div>
`;
